### YamlMime:ManagedReference
items:
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  id: FirstChanceExceptionEventArgs
  children:
  - System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.#ctor(System.Exception)
  - System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception
  langs:
  - csharp
  name: FirstChanceExceptionEventArgs
  nameWithType: FirstChanceExceptionEventArgs
  fullName: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  type: Class
  summary: "提供 managed 例外狀況第一次發生時，common language runtime 會開始搜尋的事件處理常式之前引發通知事件資料。"
  remarks: "這個類別會提供例外狀況處理常式<xref:System.AppDomain.FirstChanceException?displayProperty=fullName>具有存取權的例外狀況的事件。</xref:System.AppDomain.FirstChanceException?displayProperty=fullName>"
  example:
  - "The following example creates a series of application domains named `Child_0` through `Child_3`, with a `Worker` object in each application domain. Each `Worker` object has a reference to the `Worker` object in the next application domain, except for the `Worker` in the last application domain. The <xref:System.AppDomain.FirstChanceException> event is handled in all application domains except `Child_1`.  \n  \n When the application domains have been created, the default application domain calls the `TestException` method for the first child application domain. Each `Worker` object calls the `TestException` method for the next, until the last `Worker` throws an exception that is either handled or unhandled. Thus, the current thread passes through all the application domains, and `TestException` is added to the stack in each application domain.  \n  \n When the last `Worker` object handles the exception, the <xref:System.AppDomain.FirstChanceException> event is raised only in the last application domain. The other application domains never get a chance to handle the exception, so the event is not raised.  \n  \n When the last `Worker` object does not handle the exception, the <xref:System.AppDomain.FirstChanceException> event is raised in each application domain that has an event handler. After each event handler has finished, the stack continues to unwind until the exception is caught by the default application domain.  \n  \n> [!NOTE]\n>  To see how the stack display grows as the event is raised closer and closer to the default application domain, change `e.Exception.Message` to `e.Exception` in the `FirstChanceHandler` event handlers. Notice that when `TestException` is called across application domain boundaries, it appears twice: once for the proxy and once for the stub.  \n  \n [!code-vb[System.AppDomain.FirstChanceException#1](~/add/codesnippet/visualbasic/t-system.runtime.excepti_1.vb)]\n [!code-cs[System.AppDomain.FirstChanceException#1](~/add/codesnippet/csharp/t-system.runtime.excepti_1.cs)]"
  syntax:
    content: 'public class FirstChanceExceptionEventArgs : EventArgs'
  inheritance:
  - System.EventArgs
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.#ctor(System.Exception)
  id: '#ctor(System.Exception)'
  parent: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  langs:
  - csharp
  name: FirstChanceExceptionEventArgs(Exception)
  nameWithType: FirstChanceExceptionEventArgs.FirstChanceExceptionEventArgs(Exception)
  fullName: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.FirstChanceExceptionEventArgs(Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Runtime.ExceptionServices
  summary: "初始化的新執行個體<xref href=&quot;System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs&quot;></xref>類別具有指定的例外狀況。"
  remarks: "Common language runtime，會呼叫這個建構函式不是由應用程式程式碼。"
  syntax:
    content: public FirstChanceExceptionEventArgs (Exception exception);
    parameters:
    - id: exception
      type: System.Exception
      description: "例外狀況，只 managed 程式碼，所擲回和，將會檢驗<xref href=&quot;System.AppDomain.UnhandledException&quot;></xref>事件。"
  overload: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception
  id: Exception
  parent: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  langs:
  - csharp
  name: Exception
  nameWithType: FirstChanceExceptionEventArgs.Exception
  fullName: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception
  type: Property
  assemblies:
  - mscorlib
  namespace: System.Runtime.ExceptionServices
  summary: "在 managed 程式碼中擲回的例外狀況對應的 managed 例外狀況物件。"
  remarks: "處理的方法<xref:System.AppDomain.UnhandledException?displayProperty=fullName>事件可以檢查例外狀況，並執行適當的記錄或其他 notification services。</xref:System.AppDomain.UnhandledException?displayProperty=fullName>       如範例程式碼，請參閱<xref:System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs>.</xref:System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs>"
  syntax:
    content: public Exception Exception { get; }
    return:
      type: System.Exception
      description: "新擲回的例外狀況。"
  overload: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception*
  exceptions: []
  platform:
  - net462
references:
- uid: System.EventArgs
  isExternal: false
  name: System.EventArgs
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.#ctor(System.Exception)
  parent: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  isExternal: false
  name: FirstChanceExceptionEventArgs(Exception)
  nameWithType: FirstChanceExceptionEventArgs.FirstChanceExceptionEventArgs(Exception)
  fullName: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.FirstChanceExceptionEventArgs(Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception
  parent: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  isExternal: false
  name: Exception
  nameWithType: FirstChanceExceptionEventArgs.Exception
  fullName: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.#ctor*
  parent: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  isExternal: false
  name: FirstChanceExceptionEventArgs
  nameWithType: FirstChanceExceptionEventArgs.FirstChanceExceptionEventArgs
- uid: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs.Exception*
  parent: System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs
  isExternal: false
  name: Exception
  nameWithType: FirstChanceExceptionEventArgs.Exception
